pkg/dynamicinstrumentation/codegen/c/maps.h

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+#ifndef DI_MAPS_H
+#define DI_MAPS_H
+
+// The events map is the ringbuffer used for communicating events from
+// bpf to user space.
+struct {
+    __uint(type, BPF_MAP_TYPE_RINGBUF);
+    __uint(max_entries, 1 << 24);
+} events SEC(".maps");
+
+// The zeroval map is used to have pre-zero'd data which bpf code can
+// use to zero out event buffers (similar to memset, but verifier friendly).
+struct {
+    __uint(type, BPF_MAP_TYPE_ARRAY);
+    __uint(key_size, sizeof(__u32));
+    __uint(value_size, sizeof(char[PARAM_BUFFER_SIZE]));
+    __uint(max_entries, 1);
+} zeroval SEC(".maps");
```

Comment:
FYI we have helper macros for defining maps in `map-defs.h`, so you can do things like `BPF_ARRAY_MAP(zeroval, char[PARAM_BUFFER_SIZE], 1)`

Reply:
I've updated to use the macros in map-defs.h, and had to adapt the macros a bit to support stacks which would fail on creation if key is specified.